pkg/scheduler/metrics/metrics.go

```diff
@@ -87,6 +87,22 @@ var (
 			Name:      "total_preemption_attempts",
 			Help:      "Total preemption attempts in the cluster till now",
 		})
+
```

Comment:
Are you suggesting to add metrics for `scheduling with ecache` process, or just ecache operations (update & invalidate)? I think the former make sense.

Reply:
I think that the "latency" and "duration" metrics above are capturing this information. I guess if the scenario is that scheduler throughput declines gradually over several hours or days, then we want to know if eCache is the culprit?

Reply:
Hi, sorry I haven't responded to this recently. In looking at adding further equiv. cache metrics, I worry that latency may not measure the right thing. In particular, the hash value is computed separately from EquivalenceCache.RunPredicate. Performance concerns have been raised about hash computation, but not about the map lookups.

I think there is plenty of opportunity to improve scheduler metrics (there is no counter for pods scheduled today) so I would like to push additional metrics into other PRs.

Reply:
> In particular, the hash value is computed separately from EquivalenceCache.RunPredicate. 

Thanks. I was just curious about locking & unlocking happening at https://github.com/kubernetes/kubernetes/blob/b71966aceaa3c38040236bc0decc6fad36eeb762/pkg/scheduler/core/equivalence_cache.go#L74

But if there are no concerns around it, I am fine. My past experiences with RWLocks have not been good when it comes to scaling but I think we are not going to use them.

> Performance concerns have been raised about hash computation, but not about the map lookups.

Sure. I agree that hash computation could take time to avoid collisions. We can get that information from benchmarking too.

> I think there is plenty of opportunity to improve scheduler metrics (there is no counter for pods scheduled today) so I would like to push additional metrics into other PRs.

Fair enough.

Reply:
xref the upcoming optimization of lock here: https://github.com/kubernetes/kubernetes/issues/63784#issuecomment-389345703